packages/databyss-editor/lib/clipboardUtils.tsx

```diff
@@ -0,0 +1,445 @@
+import ReactDOMServer from 'react-dom/server'
+import ObjectId from 'bson-objectid'
+import cloneDeep from 'clone-deep'
+import { BlockType } from '@databyss-org/services/interfaces'
+import { Text, Range, Selection, EditorState, Block } from '../interfaces'
+import { isAtomicInlineType } from './util'
+import { stateToHTMLString } from './slateUtils'
+
+interface BasicBlock {
```

Comment:
when doing block manipulation (split, merge) we do not require the block `id`, the block `id` logic is handled by the functions calling `splitBlockAtOffset` `mergeBlock` `insertText`

Reply:
Ah ok. Let's make `_id` an optional field so we can use the `Block` interface here. As in
```
export interface Block {
  _id?: string
  type: BlockType
  text: Text
}
```

While we're at it, we can do the same for `Selection`, because TS is complaining sometimes that `_id` is missing.